docs/specs/#1984 - Apps and Packages Spec/spec.md

```diff
@@ -0,0 +1,118 @@
+| 3   |    A user can select multiple applications at a time to perform bulk actions like update or uninstall       |    0      |
+| 4   |   A user can search to find an application       |      0    |
+| 5   |     A user can filter the page by update available      |      0    |
+| 6   |    A user can sort the page by application name, publisher, size, and source.      |    0      |
+
+## 3. User Stories
+
+### 3.1 User story - Updating and Rolling Back Packages
+
+A software developer wants to update a package to the latest version but finds that it has a bug affecting their system.
+
+#### Job-to-be-done
+
+1. Update a package to the latest version
+2. Roll back to a previous version of a package
+
+#### User experience
+
+- A software developer launches Dev Home and navigates to the Apps & Packages page.
+- They select the package that they want to update and click the Update button.
+- They are prompted to confirm the update and clicks Yes. The package is updated to the latest version.
+- They later find that the update has a security bug, so they go back to Dev Home to quickly revert to a previous version.
+- They select the package they want to roll back and click on the version.
+- They can see the available versions and select the version they want to roll back to.
+- The package is rolled back to the previous version.
+
+### 3.2 User story - Performing Bulk Actions
+
+A software developer wants to clean up space on their machine and uninstall large packages they are no longer using.
+
+#### Job-to-be-done
+
+1. Sort packages by size  
+2. Select multiple packages to uninstall in bulk
+
+#### User experience
+
+- User launches Dev Home and navigates to the Apps & Packages page.
+- They click on the Size column to sort the packages by size.
+- They select 3 packages they want to uninstall and click the Uninstall button.
+- They are prompted to confirm the uninstall and click Yes. All 3 packages are uninstalled.
+
+## 4. Requirements
+
+### 4.1 Functional Requirements
+
+#### Summary
+
+This feature specification outlines the addition of a unified package management interface in Dev Home. It allows users to manage apps & packages from various package managers in a single, streamlined interface. With advanced features such as dependency and version management, batch operations, and rollback capabilities, the new interface provides a comprehensive package management solution. The result is a more efficient and user-friendly package management experience that substantially simplifies the workflow for software developers and IT professionals.
+
+#### Detailed Functional Requirements
+
+| No. | Requirement | Pri |
+| --- | ----------- | --- |
+| 1   |     The list of applications shown are winget or store applications (source: winget, msstore from winget list)      |  0   |
+| 2   |     The management experience allows updating, version management, and uninstallation of winget or msstore packages.          |  0   |
```

Comment:
Naive question: does store allow for installing not-the-latest package?

Reply:
no, store supports only the latest version, and does not show the version either. Store will let us update and uninstall.

Reply:
Can we update the spec (items 2 and 3 here) to reflect that?

Reply:
done